### PR TITLE
distsql: use a background goroutine to decode bytes read from tikv/mocktikv

### PR DESCRIPTION
**Pull request by zz-jason**

### CODE REVIEW

**Comment by zz-jason** [comment]
@winoros @shenli PTAL

File: distsql/distsql.go [hunk 1]
```diff
@@ -153,25 +162,80 @@ func (r *selectResult) NextRaw(ctx context.Context) ([]byte, error) {
...
+	go r.decode(ctx)
+}
+
+func (r *selectResult) decode(ctx context.Context) {
+	defer close(r.dstChan)
+	var resource *decodedChk = nil
+	var ok bool = false
+	for {
+		select {
+		case resource, ok = <-r.srcChan:
```
Comment (coocood) [inline distsql/distsql.go]: No one closes the `srcChan`, so we don't need to check `ok`.

**Comment by zz-jason** [comment]
This pr may reduce the performance in some situation, and since we are going to support Chunk in tikv, this PR is meaningless.

**Pull request closed without merging**